cluster.c

```diff
@@ -759,20 +814,24 @@ RRStatus ShardingInfoAddShardGroup(RedisRaftCtx *rr, ShardGroup *new_sg)
     ShardingInfo *si = rr->sharding_info;
 
     /* Validate first */
-    if (ShardingInfoValidateShardGroup(rr, new_sg) != RR_OK)
-        return RR_ERROR;
-
-    si->shard_groups_num++;
-    ShardGroup *sg = RedisModule_Alloc(sizeof(ShardGroup));
-    if (RedisModule_DictSetC(si->shard_group_map, new_sg->id, strlen(new_sg->id), sg) != REDISMODULE_OK) {
+    if (ShardingInfoValidateShardGroup(rr, new_sg) != RR_OK) {
         return RR_ERROR;
     }
-    if (strlen(new_sg->id) >= sizeof(sg->id)) {
+
+    if (strlen(new_sg->id) >= sizeof(new_sg->id)) {
         return RR_ERROR;
     }
+    ShardGroup *sg = RedisModule_Calloc(1, sizeof(ShardGroup));
```

Comment:
Similar to my previous comment, we need to call free this allocation if we goto `error` section?